apps/files_sharing/lib/Controller/ExpiredShareAPIController.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2016 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Files_Sharing\Controller;
+
+use OCA\Files_Sharing\ResponseDefinitions;
+use OCP\App\IAppManager;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\NoAdminRequired;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\Files\IRootFolder;
+use OCP\IDateTimeZone;
+use OCP\IGroupManager;
+use OCP\IL10N;
+use OCP\IPreview;
+use OCP\IRequest;
+use OCP\IURLGenerator;
+use OCP\IUserManager;
+use OCP\Share\IManager as ShareManager;
+use OCP\Share\IShare;
+use OCP\UserStatus\IManager as UserStatusManager;
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @psalm-import-type Files_SharingShare from ResponseDefinitions
+ */
+class ExpiredShareAPIController extends ShareApiControllerFactory {
+
+	public function __construct(
+		IRequest $request,
+		protected ShareManager $shareManager,
+		protected string $userId,
+		protected IUserManager $userManager,
+		protected IGroupManager $groupManager,
+		protected IRootFolder $rootFolder,
+		protected IAppManager $appManager,
+		protected ContainerInterface $serverContainer,
+		protected UserStatusManager $userStatusManager,
+		protected IPreview $previewManager,
+		protected IDateTimeZone $dateTimeZone,
+		protected IURLGenerator $urlGenerator,
+		protected IL10N $l,
+		protected LoggerInterface $logger,
```

Comment:
```suggestion
		ShareManager $shareManager,
		string $userId,
		IUserManager $userManager,
		IGroupManager $groupManager,
		IRootFolder $rootFolder,
		IAppManager $appManager,
		ContainerInterface $serverContainer,
		UserStatusManager $userStatusManager,
		IPreview $previewManager,
		IDateTimeZone $dateTimeZone,
		IURLGenerator $urlGenerator,
		IL10N $l,
		LoggerInterface $logger,
```
You are redeclaring properties from parent class, it’s better to simply let the parent class declare them and simply pass the parameters to the constructor.